src/Microsoft.Health.Dicom.Blob/Features/ExternalStore/ExternalBlobClient.cs

```diff
@@ -62,12 +77,67 @@ public BlobContainerClient BlobContainerClient
                         }
                         catch (Exception ex)
                         {
-                            throw new DataStoreException(ex, isExternal: IsExternal);
+                            throw new DataStoreException(ex, isExternal: true);
                         }
                     }
                 }
             }
             return _blobContainerClient;
         }
     }
+
+    private static string SanitizeServiceStorePath(string path)
+    {
+        return !path.EndsWith("/", StringComparison.OrdinalIgnoreCase) ? path + "/" : path;
+    }
+
+    /// <summary>
+    /// See https://learn.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#blob-names
+    /// </summary>
+    private void ServiceStorePathContainsInvalidCharactersCheck()
+    {
+        // Reserved URL characters must be properly escaped.
+        // https://www.rfc-editor.org/rfc/rfc3986#section-2.2
+        // reserved    = gen-delims / sub-delims
+        // gen-delims  = ":" / "/" / "?" / "#" / "[" / "]" / "@"
+        // sub-delims  = "!" / "$" / "&" / "'" / "(" / ")"
+        //               / "*" / "+" / "," / ";" / "="
+        if (_externalStoreOptions.StorageDirectory.
+            Select(x => !char.IsLetterOrDigit(x) && !_allowedChars.Contains(x)).
+            Any(x => x is true))
+        {
+            throw new DataStoreException(DicomCoreResource.ExternalDataStoreInvalidCharactersInServiceStorePath, isExternal: true);
+        }
+    }
+
+    /// <summary>
+    /// See https://learn.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#blob-names
+    /// </summary>
+    private void ServiceStorePathSegmentInvalidCheck()
+    {
+        // If your account does not have a hierarchical namespace, then the number of path segments comprising the blob
+        // name cannot exceed 254. A path segment is the string between consecutive delimiter characters
+        // (e.g., the forward slash '/') that corresponds to the name of a virtual directory.
+        if (_externalStoreOptions.StorageDirectory.Count(c => c == '/') > MaxBlobSegmentsAllowed)
+        {
+            throw new DataStoreException(DicomCoreResource.ExternalDataStoreInvalidServiceStorePathSegments, isExternal: true);
+        }
+    }
+
+    /// <summary>
+    /// This validation necessary only for OSS as the path will be specified for managed services.
+    /// </summary>
+    /// <exception cref="DataStoreException"></exception>
+    private void EnsureValidConfiguration()
```

Comment:
We can catch this earlier on start-up, right? I think we could make `ExternalBlobDataStoreConfiguration` implement 
[IValidatableObject](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.ivalidatableobject?view=net-7.0) like some other classes used with `IOptions<T>` to fail server start up when misconfigured

Reply:
I can understand not wanting to fail the server for runtime issues, but shouldn't we validate some of these properties ahead of time, like via ARM and/or portal? So that way we're not validating once your service is up-and-running?